File: android/src/main/java/com/gaya/digital_turbine_plugin/BannerViewFactory.java
package com.gaya.digital_turbine_plugin;

import android.app.Activity;
import android.content.Context;
import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;
import java.util.Map;

public class BannerViewFactory extends PlatformViewFactory {
    private final BinaryMessenger messenger;
    private final Activity activity;

    public BannerViewFactory(BinaryMessenger messenger, Activity activity) {
        super(StandardMessageCodec.INSTANCE);
        this.messenger = messenger;
        this.activity = activity;
    }

    @Override
    public PlatformView create(Context context, int viewId, Object args) {
        Map<String, Object> creationParams = (Map<String, Object>) args;
        return new BannerView(context, activity, viewId, creationParams, messenger);
    }
}

File: android/src/main/kotlin/com/gaya/digital_turbine_plugin/BannerViewFactory.kt
package com.gaya.digital_turbine_plugin

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import android.app.Activity

class BannerViewFactory(
    private val messenger: BinaryMessenger,
    private val activity: Activity
) : io.flutter.plugin.platform.PlatformViewFactory(io.flutter.plugin.common.StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val creationParams = args as? Map<String, Any>
        return BannerViewWrapper(context, activity, viewId, creationParams, messenger)
    }
}